src/function_greed_non_linear.py

```diff
@@ -171,6 +171,10 @@ def __init__(self, in_features, out_features, opt, data, device, bias=False):
         self.W_W = Parameter(torch.Tensor(in_features, in_features - 1), requires_grad=opt['gnl_W_param_free'])
         self.t_a = Parameter(torch.Tensor(in_features), requires_grad=opt['gnl_W_param_free'])
         self.r_a = Parameter(torch.Tensor(in_features), requires_grad=opt['gnl_W_param_free'])
+        if self.time_dep_w:
+          self.W_W = Parameter(torch.Tensor(self.num_timesteps, in_features, in_features - 1), requires_grad=opt['gnl_W_param_free'])
```

Comment:
For the case "diag_dom" (i) Agnostic approach. W is only a function of t through parameters t_a and r_a, so I don't think we need an extra dimension here